lib/agent.js

```diff
@@ -111,6 +121,8 @@ Agent.prototype._doClose = function() {
   for (var k in this._msgIdToReq)
     this._msgIdToReq[k].sender.reset()
 
+  if (this._opts.socket) return;
```

Comment:
[CS] remove ";"
[CS] the return statement should be placed into a separate line